jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +172,81 @@ public void removeComment(final String serverTemplateId, final String containerI
         return comparing(CaseMilestoneSummary::getStatus).thenComparing(request.getSortByAsc() ? comparatorByName: comparatorByName.reversed());
     }
 
+    @Override
+    public Actions getCaseActions(String serverTemplateId, String container, String caseId, String userId) {
+        Actions actions = new Actions();
+        actions.setAvailableActions(getAdHocActions(serverTemplateId, container, caseId));
+        actions.setInProgressAction(getInProgressActions(container, caseId));
+        actions.setCompleteActions(getCompletedActions(container, caseId));
+        return actions;
+    }
+
+    public List<CaseActionSummary> getInProgressActions(String containerId, String caseId) {
+        List<NodeInstance> activeNodes = client.getActiveNodes(containerId, caseId, 0, PAGE_SIZE_UNLIMITED);
+
+        return activeNodes.stream()
+                .map(new CaseActionNodeInstanceMapper())
+                .map(s -> {
```

Comment:
I would try to avoid calling `.map` twice. My suggestion would be to pass actualOwner and actionType as constructor parameters instead. That way you can also avoid having to map the workItemId. You would have access to it inside the map fragment.